spinn_utilities/make_tools/file_convertor.py

```diff
@@ -0,0 +1,453 @@
+FORMATEXP = re.compile("%\d*(?:\.\d+)?[cdiksuxR]")
+LOG_END_REGEX = re.compile('\)(\s)*;')
+END_COMMENT_REGEX = re.compile("/*/")
+LOG_START_REGEX = re.compile(
+    "log_((info)|(error)|(debug)|(warning))(\s)*\(")
+
+# Status values
+NORMAL_CODE = 0
+COMMENT = NORMAL_CODE + 1
+IN_LOG = COMMENT + 1
+IN_LOG_CLOSE_BRACKET = IN_LOG + 1
+
+MINIS = {"log_info(": "log_mini_info(",
+         "log_error(": "log_mini_error(",
+         "log_debug(": "log_mini_debug(",
+         "log_warning(": "log_mini_warning("}
+
+LEVELS = {"log_info(": "[INFO]",
+          "log_error(": "[ERROR]",
+          "log_debug(": "[DEBUG]",
+          "log_warning(": "[WARNING]"}
+
+MAX_LOG_PER_FILE = 100
+
+
+class FileConvertor(object):
+    # __slots__ = [
+    #    "_dest", "_dest_basename", "_src", "_src_basename"]
+
+    # REGEXP = re.compile("%(?:\d+\$)?[dfsu]")
+
+    def __init__(self, src, dest, dict, range_start):
+        self._src = os.path.abspath(src)
+        self._dest = os.path.abspath(dest)
+        self._message_id = range_start
+        self._dict = dict
+
+    def run(self):
+        if not os.path.exists(self._src):
+            raise Exception("Unable to locate source {}".format(src))
+        dest_dir = os.path.dirname(os.path.realpath(self._dest))
+        if not os.path.exists(dest_dir):
+            os.makedirs(dest_dir)
+        self.convert_c()
+        return self._message_id
+
+    def convert_c(self):
+        with open(self._src) as src_f:
+            with open(self._dest, 'w') as dest_f:
+                dest_f.write(
+                    "// DO NOT EDIT! THIS FILE WAS GENERATED FROM {}\n\n"
+                    .format(self.unique_src()))
+                self._too_many_lines = 2
+                self._status = NORMAL_CODE
+                for self._line_num, self._text in enumerate(src_f):
+                    if self._too_many_lines > 0:
+                        # Try to recover the lines added by do not edit
+                        check = self._text.strip()
+                        if len(check) == 0 or check == "*":
+                            self._too_many_lines -= 1
+                            continue
+                    if not self._process_line(dest_f):
+                        self._comment_start = 0
+                        self._process_chars(dest_f)
+        # print (self._dest)
+        return self._message_id
+
+    def _process_line(self, dest_f):
+        if self._status == COMMENT:
+            return self._process_line_in_comment(dest_f)
+        if "/*" in self._text:
+            return self._process_line_comment_start(dest_f)
+
+        if self._status == IN_LOG:
+            return self._process_line_in_log(dest_f)
+
+        if self._status == IN_LOG_CLOSE_BRACKET:
+            return self._process_line_in_log_close_bracket(dest_f)
+
+        assert self._status == NORMAL_CODE
+        return self._process_line_normal_code(dest_f)
+
+    def _check_token(self):
```

Comment:
This whole method isnt used anywhere that i can find (and wouldnt work for a function definition).

Reply:
nuked dead code that was confusing Alan.